containers/ddev-dbserver/Makefile

```diff
@@ -46,43 +48,62 @@ build: container
 		set -euo pipefail ; \
 		export baseversion=$$(echo $$item | awk -F ':' '{print $$1}'); \
 		export fullversion=$$(echo $$item | awk -F ':' '{print $$2}'); \
 		printf "\n\n========== Building MySQL $${baseversion} with image $${fullversion} ==========\n", ${baseversion}, ${fullversion}; \
-		docker build $(DOCKER_ARGS) --build-arg="DBVERSION=$$baseversion" --build-arg="PARENTIMAGEVERSION=$$fullversion" --build-arg="DBTYPE=mysql" -t "drud/ddev-dbserver-mysql-$${baseversion%%%.*}:$(VERSION)" . ; \
+		docker buildx build -o type=docker $(DOCKER_ARGS) --progress=$(PROGRESS) --build-arg="DBVERSION=$$baseversion" --build-arg="PARENTIMAGEVERSION=$$fullversion" --build-arg="DBTYPE=mysql" -t "drud/ddev-dbserver-mysql-$${baseversion%%%.*}:$(VERSION)" . ; \
 	done
 
-push: container
-	for item in $(MARIADB_VERSIONS); do \
+# Below has mariadb_containers and mysql_containers as a prerequisite because those build the amd64 images.
+# So, the multi-arch approach here is a bit different from the ones in the other Makefiles.
+multi-arch: prep mariadb_containers mysql_containers
+	for item in $(MARIADB_VERSIONS_arm64); do \
+		set -euo pipefail ; \
+		export baseversion=$$(echo $$item | awk -F ':' '{print $$1}'); \
+		export fullversion=$$(echo $$item | awk -F ':' '{print $$2}'); \
+		printf "\n\n========== Building MariaDB $${baseversion} with image $${fullversion} ==========\n"; \
+		docker buildx build --platform=linux/arm64 $(DOCKER_ARGS) --progress=$(PROGRESS) --build-arg="DBVERSION=$${baseversion}" --build-arg="PARENTIMAGEVERSION=$$fullversion" --build-arg="DBTYPE=mariadb" -t "drud/ddev-dbserver-mariadb-$${baseversion}:$(VERSION)" . ; \
+	done	
+
+push: prep
+	for item in $(MARIADB_VERSIONS_amd64); do \
 		set -euo pipefail ;\
 		export baseversion=$$(echo $$item | awk -F ':' '{print $$1}'); \
-		docker push "drud/ddev-dbserver-mariadb-$${baseversion%%%.*}:$(VERSION)"; \
+		export fullversion=$$(echo $$item | awk -F ':' '{print $$2}'); \
+		export archs=$$(if [[ "$(MARIADB_VERSIONS_arm64)" == *"$$item"* ]]; then echo "linux/amd64,linux/arm64"; else echo "linux/amd64"; fi);\
+		docker buildx build --push --platform $${archs} $(DOCKER_ARGS) --progress=$(PROGRESS) --build-arg="DBVERSION=$${baseversion}" --build-arg="PARENTIMAGEVERSION=$$fullversion" --build-arg="DBTYPE=mariadb" -t "drud/ddev-dbserver-mariadb-$${baseversion}:$(VERSION)" .; \
 	done
-	for item in $(MYSQL_VERSIONS) ; do \
+	for item in $(MYSQL_VERSIONS_amd64) ; do \
 		set -euo pipefail ;\
 		export baseversion=$$(echo $$item | awk -F ':' '{print $$1}'); \
-		docker push "drud/ddev-dbserver-mysql-$${baseversion%%%.*}:$(VERSION)"; \
+		export fullversion=$$(echo $$item | awk -F ':' '{print $$2}'); \
+		docker buildx build --push --platform linux/amd64 $(DOCKER_ARGS) --progress=$(PROGRESS) --build-arg="DBVERSION=$$baseversion" --build-arg="PARENTIMAGEVERSION=$$fullversion" --build-arg="DBTYPE=mysql" -t "drud/ddev-dbserver-mysql-$${baseversion%%%.*}:$(VERSION)" .; \
```

Comment:
This file has become quite a mess, but it works 😅 let me know if you have any feedback on this approach and if you'd do some things differently